src/download_data.py

```diff
@@ -0,0 +1,46 @@
+# author: Group 24
+# date: 2021-11-19
+
+'''This script downloads data from 'wosaku/crime-in-vacouver' Kaggle.com using
+its API and saves its to a data file path.
+from a csv file. This script takes an unquoted file path.
+
+Usage: download_data.py --file_path=<file_path>
+
+Options:
+--file_path=<file_path>   Path to the data file
+'''
+import os
+import pandas as pd
+from docopt import docopt
+
+opt = docopt(__doc__)
+
+
+try:
+    import kaggle
+except ModuleNotFoundError:
+    print('ERROR : Please install the kaggle module using : pip install kaggle')
+except OSError:
+    print ("ERROR :You must authenticate using an API token please follow the instructions")
+    print("First, create a kaggle account. Go to account setting and create a 'New API Token'")
+    print("then, move the downloaded .json file to your ~/.kaggle directory in your computer")
+
+
+def main(file_path):
+    
+    
+    # if the files is not in the directory create one and dowload it
+    if not os.path.exists(os.path.join(file_path, 'data\crime.csv')):

```

Comment:
When I tested it with `python src/download_data.py --file_path="data/raw"`, the file actually is created under `data/raw/data` which is unexpected. Suggest modifying the code as below to further generalize the download path:

```python
    # if the files is not in the directory create one
    if not os.path.exists(os.path.dirname(file_path)):

        # create a data directory
        os.makedirs(os.path.dirname(file_path))
        
    # download the data in that directory
    kaggle.api.dataset_download_files(kaggle_path,
                                      path=file_path,
                                      unzip=True)
```

Reply:
i think this is not considering the files inside data

Reply:
Right, as I assume we always overwrite the file. What do you think?

Reply:
my reference to the assumption:
https://github.com/UBC-MDS/contraceptive_method_predictor/blob/main/src/download_data.py